src/components/ButtonGroup/ButtonGroup.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import glamorous from 'glamorous';
+
+import Button from '../Button/Button';
+import constants from '../../constants';
+
+const { borderRadius } = constants;
+
+const ButtonGroupContainer = glamorous.div({
+   display: 'inline-block',
+});
+
+const GroupedButton = glamorous(Button)({
+   borderRadius: 0,
+
+   '&:first-of-type': {
+      borderTopLeftRadius: borderRadius,
+      borderBottomLeftRadius: borderRadius,
+   },
+   '&:last-of-type': {
+      borderTopRightRadius: borderRadius,
+      borderBottomRightRadius: borderRadius,
+   },
+});
+
+const ButtonGroup = ({
+   children,
+   className,
+   disabled,
+   onMouseEnter,
+   onMouseLeave,
+}) => (
+   <ButtonGroupContainer
+      className={className}
+      onMouseEnter={onMouseEnter}
+      onMouseLeave={onMouseLeave}
+   >
+      {React.Children.map(children, child => {
+         if (child.type !== Button) {
+            return child;
+         }
+         return <GroupedButton disabled={disabled} {...child.props} />;
+      })}
+   </ButtonGroupContainer>
+);
+
+ButtonGroup.propTypes = {
+   /** Buttons to be grouped together. */
+   children: PropTypes.arrayOf(PropTypes.node),
+   /** Set class name of containing element. */
+   className: PropTypes.string,
+   /** Indicates that the control is not available for interaction */
+   disabled: PropTypes.bool,
+   /** Callback when mouse enters component. */
+   onMouseEnter: PropTypes.func,
+   /** Callback when mouse leaves component. */
+   onMouseLeave: PropTypes.func,
+};
+
+ButtonGroup.defaultProps = {
+   children: [],
```

Comment:
Does it make sense to have an empty `ButtonGroup`? I'm cool with making `children` a required prop.